instrumentation/delayed_job/lib/opentelemetry/instrumentation/delayed_job/instrumentation.rb

```diff
@@ -27,6 +27,8 @@ class Instrumentation < OpenTelemetry::Instrumentation::Base
         private
 
         def gem_version
+          # Version is hardcoded in the gemspec
+          # https://github.com/collectiveidea/delayed_job/blob/master/delayed_job.gemspec#L16
           Gem.loaded_specs['delayed_job'].version
```

Comment:
Did we want to rub some try (`&.`) on this? Or leave as-is for an exception instead of a nil?

Or maybe …

```suggestion
          Gem.loaded_specs.fetch('delayed_job').version
```

For a _different_ exception? `KeyError (key not found: "delayed_job")` as opposed to `NoMethodError (undefined method 'version' for nil:NilClass)`

Reply:
Or even … 
```suggestion
          Gem
            .loaded_specs
            .fetch('delayed_job') { raise KeyError.new("'delayed_job' not
 found among Gem.loaded_specs") }
            .version
```

… for `KeyError ('delayed_job' not found among Gem.loaded_specs)`.

🤔 Maybe that's too much fetch?

Reply:
That would trigger the `error_handler`, is that how we want to control the flow? 

https://github.com/open-telemetry/opentelemetry-ruby/blob/main/instrumentation/base/lib/opentelemetry/instrumentation/registry.rb#L84

Reply:
I think that's how flow is controlled with the existing implementation.

```ruby
Gem.loaded_specs['delayed_job'].version
```

 If it can't find `'delayed_job'` in `Gem.loaded_specs`, a`NoMethodError (undefined method 'version' for nil:NilClass)` exception would be thrown (right?) which I think is less clear about what the true error is.